src/App.js

```diff
@@ -199,144 +180,98 @@ class App extends React.Component {
       }
     }
     if ((result[result.length - 1] || "").includes(".")) {
-      console.log(result[result.length - 1]);
       if ("0123456789".includes(button)) {
         return result + button;
       }
     }
 
-       //🪲 72.2 + doesn't work FIXED
-    if ((result || "").includes('.')) {
-
+    if ((result || "").includes(".")) {
       if ("+-*/0123456789".includes(button)) {
         return result + button;
       }
     }
-
-    // 🪲 72(7+2 returns "not yet coded"
-
-
-    // operator + operator -> replace the first operator
-
-    // "" - - -> is not possible
-    //return "not yet coded";
-
     return result;
   };
 
   closeParens = (result) => {
     var numberOfOpenP = (result.match(/\(/g) || []).length;
     var numberOfCloseP = (result.match(/\)/g) || []).length;
-    //this solution works but only if there is one parenthesis missing,
-    //instead, account for all parens missing
 
     while (numberOfOpenP > numberOfCloseP) {
       result = result + ")";
       numberOfOpenP--;
     }
-    return result
+    return result;
   };
 
   getLastChar = (from) => {
     return from.slice(-1);
   };
 
-
   roundedResult = (expression) => {
     const digitAfterComma = 13;
     round(evaluate(expression), digitAfterComma);
   };
 
   addMultiplier = (expression) => {
-    //Find all pattern Digit + (
-    //expression ="1(2+6)+23(6+9)+(2+3)9"
-
     const digitAndP = /([0123456789])(\()/g;
     const pAndDigit = /(\))([0123456789])/g;
     // )( -> )*(
     const pAndp = /(\))(\()/g;
 
-    //return the value
     return expression
       .replace(digitAndP, "$1*$2")
       .replace(pAndDigit, "$1*$2")
       .replace(pAndp, "$1*$2");
-
-
   };
 
-
-
   calculate = () => {
     const { result, operate, originalLastNum } = this.state;
     let finalResult = 0;
     var tempResult = result;
 
-    // Special case with expression ending with operator, after CE use
     if ("-+*/".includes(this.getLastChar(result))) {
       tempResult = result.slice(0, -1);
     }
 
-
     tempResult = this.closeParens(tempResult) + "";
     tempResult = this.addMultiplier(tempResult) + "";
 
+    var tempResultString = tempResult;
+    const reg = /\(([0123456789/*-+.]*)\)/g;
+    var parenthesisToCalculate = tempResultString.match(reg) || [];
+    while (parenthesisToCalculate.length > 0) {
+      for (var i = 0; i < parenthesisToCalculate.length; i++) {
+        let expression = parenthesisToCalculate[i];
+        try {
+          let tempResult = round(evaluate(expression), 13);
+          tempResultString = tempResultString.replace(expression, tempResult);
+        } catch (error) {}
+      }
+      parenthesisToCalculate = tempResultString.match(reg) || [];
+    }
+    try {
+      finalResult = round(evaluate(tempResultString), 13);
+    } catch (error) {}
```

Comment:
Why was this moved out of the else block? How did you gain confidence that this doesn't have any unintended consequences?

Reply:
That's a good point, I didn't think it needed to be in an `if` and `else`, but since it may mutate code that may not show up in the tests I actually just fixed it so that it is back in an `if` and `else` block and have the tests passing. Before when I tried to preserve that structure the tests were not passing. Although to answer your question more directly, I had thought that even removing the if else wouldn't be an issue because it wouldn't hit that case here `if (matches.length > 1 && matches[0] === tempResult)` unless it was supposed to.

Reply:
And like you said, I don't want to be changing the code's behavior in a refactor so I will keep it in this if else.

Reply:
This is a bug fix, so far, not a refactor. You have to change behavior but it should be limited to only the code that fixes the bug.